### PR TITLE
fix(executor): route legacy SDK calls through gateway

### PR DESCRIPTION
**Pull request by daryllimyt**
## Summary

- keep `RegistryContext` focused on execution identity by removing action gateway socket plumbing
- have current registry SDK clients privately select the action gateway UDS from the executor environment
- patch legacy pinned registry SDK clients inside `minimal_runner` so old workflow locks route SDK calls through the gateway instead of `/internal` on the API service

## Tests

- `uv run pytest tests/unit/executor/test_minimal_runner_protocol.py tests/registry/test_context.py tests/registry/test_tracecat_client.py -q`
- `uv run ruff check tracecat/executor/minimal_runner.py packages/tracecat-registry/tracecat_registry/context.py packages/tracecat-registry/tracecat_registry/sdk/client.py tests/unit/executor/test_minimal_runner_protocol.py tests/registry/test_context.py tests/registry/test_tracecat_client.py`
- `uv run ruff format --check tracecat/executor/minimal_runner.py packages/tracecat-registry/tracecat_registry/context.py packages/tracecat-registry/tracecat_registry/sdk/client.py tests/unit/executor/test_minimal_runner_protocol.py tests/registry/test_context.py tests/registry/test_tracecat_client.py`
- `uv run basedpyright tracecat/executor/minimal_runner.py packages/tracecat-registry/tracecat_registry/context.py packages/tracecat-registry/tracecat_registry/sdk/client.py tests/unit/executor/test_minimal_runner_protocol.py tests/registry/test_context.py tests/registry/test_tracecat_client.py`

---
## Summary by cubic
Routes legacy registry SDK calls through the Action Gateway UDS from the executor to avoid hitting the API `/internal` endpoint and stabilize legacy workflows. Removes gateway plumbing from `RegistryContext`; current SDKs auto-select the socket via env, and the executor shims only legacy clients at runtime.

- **Bug Fixes**
  - In `minimal_runner`, install a transport shim that patches only legacy `TracecatClient` (no `_request_url_and_transport`) to send requests to `http://tracecat-action-gateway/internal/*` over the UDS when `TRACECAT__ACTION_GATEWAY_SOCKET` is set; current clients are untouched.
  - Preserve auth headers and timeouts; treat empty `TRACECAT__ACTION_GATEWAY_SOCKET` as disabled and fall back to the API URL.
  - Keep the `action_gateway_socket` keyword on `TracecatClient` as a compatibility override; add tests for env routing (including empty), keyword support, no-op behavior on current clients, and legacy `RegistryContext`/UDF flows in the executor.

- **Migration**
  - `RegistryContext` no longer accepts `action_gateway_socket`; transport config is removed from context.
  - `TracecatClient` reads `TRACECAT__ACTION_GATEWAY_SOCKET` automatically and still accepts `action_gateway_socket` for backward compatibility. No workflow changes required; ensure the executor sets `TRACECAT__ACTION_GATEWAY_SOCKET`.

<sup>Written for commit a0ea9c5b689dc7a8d52545ed5b1a60ac0fa54771. Summary will update on new commits. <a href="https://cubic.dev/pr/TracecatHQ/tracecat/pull/2722?utm_source=github">Review in cubic</a></sup>

### CODE REVIEW

File: packages/tracecat-registry/tracecat_registry/sdk/client.py [hunk 1]
```diff
@@ -41,7 +40,6 @@ def __init__(
         self,
         *,
         api_url: str | None = None,
-        action_gateway_socket: str | None = None,
         token: str | None = None,
         workspace_id: str | None = None,
         timeout: float = 120.0,
```
[the remark this answers is not in the document]
Comment (daryllimyt) [reply packages/tracecat-registry/tracecat_registry/sdk/client.py:46]: Rechecked this after the deployment context: the keyword existed in the rc16/rc17/rc18 line, and rc17 was deployed in EU. I kept `TracecatClient(action_gateway_socket=...)` as a narrow compatibility alias while keeping `RegistryContext` transport-free. The executor legacy SDK shim still handles old cached registry artifacts, and `test_action_gateway_socket_keyword_is_supported_for_compat` covers the explicit keyword path.

**Comment by daryllimyt** [comment]
@cubic review

**Comment by daryllimyt** [comment]
@codex review
